ext/storage_sources/s3_store/s3_connection.cpp

```diff
@@ -11,51 +11,78 @@
 
 /*
  * ListBuckets --
- *     Builds a list of buckets from AWS account into a vector. Returns true if success, otherwise
- *     false.
+ *     Builds a list of buckets from AWS account into a vector. Returns 0 if success, otherwise
+ *     1.
  */
-bool
+int
 S3Connection::ListBuckets(std::vector<std::string> &buckets) const
 {
     auto outcome = m_S3CrtClient.ListBuckets();
     if (outcome.IsSuccess()) {
         for (const auto &bucket : outcome.GetResult().GetBuckets())
             buckets.push_back(bucket.GetName());
-        return true;
+        return (0);
     } else {
         std::cerr << "Error in ListBuckets: " << outcome.GetError().GetMessage() << std::endl
                   << std::endl;
-        return false;
+        return (1);
     }
 }
 
 /*
  * ListObjects --
- *     Builds a list of object names from a S3 bucket into a vector. Returns true if success,
- *     otherwise false.
+ *     Builds a list of object names from an S3 bucket into a vector. Returns 0 if success,
+ * otherwise 1.
  */
-bool
-S3Connection::ListObjects(const std::string &bucketName, std::vector<std::string> &objects) const
+int
+S3Connection::ListObjects(const std::string &bucketName, const std::string &prefix,
```

Comment:
I think we want the bucket name passed to the `S3Connection` constructor and saved in a class member variable, correct?

Reply:
That is correct, but it is okay for that change not to be a part of this ticket. The bucket name will be passed from the python test all the way to the class instantiation. I am working on that change to happen.